src/app/dashboard/item/[id]/page.tsx

```diff
@@ -0,0 +1,27 @@
+import { getItemDetails } from '@/lib/models/item'
+import { getCurrentAuthUser } from '@/lib/models/user'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import { Box, Button, Typography } from '@mui/material'
+
+export default async function Page({ params }: { params: { id: string } }) {
+  const authUser = getCurrentAuthUser()
+  const itemDetails = await getItemDetails(authUser, params.id)
```

Comment:
Because this is a component rendered on the server, we can do things like get the authenticated user and call our database right in the component 😮 

It seems a little scary, but it is secure. To validate this, you can try adding a `use client` to the top of this file, and it will throw an error because these functions can only be performed on the server.